src/pperm.cc

```diff
@@ -6669,10 +6321,10 @@ static Int InitKernel(StructInitInfo * module)
     PowFuncs[T_PPERM2][T_PPERM4] = PowPPerm24;
     PowFuncs[T_PPERM4][T_PPERM2] = PowPPerm42;
     PowFuncs[T_PPERM4][T_PPERM4] = PowPPerm44;
-    QuoFuncs[T_PPERM2][T_PERM2] = QuoPPerm2Perm2;
-    QuoFuncs[T_PPERM4][T_PERM4] = QuoPPerm4Perm4;
-    QuoFuncs[T_PPERM2][T_PERM4] = QuoPPerm2Perm4;
-    QuoFuncs[T_PPERM4][T_PERM2] = QuoPPerm4Perm2;
+    QuoFuncs[T_PPERM2][T_PERM2] = QuoPPermPerm;
+    QuoFuncs[T_PPERM4][T_PERM4] = QuoPPermPerm;
+    QuoFuncs[T_PPERM2][T_PERM4] = QuoPPermPerm;
+    QuoFuncs[T_PPERM4][T_PERM2] = QuoPPermPerm;
```

Comment:
I just learned that `INV` for T_PERM2/4 is actually guaranteed to return something in the same rep, so I guess we could optimize this code again by calling `ProdPPerm2/4Perm2/4` directly, instead of `PROD`. But will save just a few cycles to load the function pointer from memory (memory that likely is in the cache anyway), so I am not sure it's worth it? (The extract functions of course then also would in turn require more cache, so I don't dare predict whether it would have positive, negative or no effect at all...)

So unless somebody is really concerned about this, I'd leave it at this single generic function for now...

As a matter of fact, I just noticed that we can even get rid of this completely, in which case `QuoDefault` will be used, which does the right thing anyway.